atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleQueryFactory.java

```diff
@@ -45,10 +45,7 @@ public OracleQueryFactory(OracleDdlConfig config, String tableName, boolean hasO
 
     @Override
     public FullQuery getLatestRowQuery(byte[] row, long ts, ColumnSelection columns, boolean includeValue) {
-        String query = " /* GET_LATEST_ONE_ROW_INNER (" + tableName + ") */ "
-                + " SELECT"
-                + "   /*+ USE_NL(t m) LEADING(t m) CARDINALITY(t 1) CARDINALITY(m 10) INDEX(m "
-                + PrimaryKeyConstraintNames.get(tableName) + ") */ "
+        String query = "SELECT"
```

Comment:
Most of the hints here are simply wrong - there is no table alias t.

More generally:

1) Hints of the form INDEX(alias PrimaryKeyConstraintNames.get(tableName)) are redundant because in the Oracle implementation of DBKVS, all these tables are Index Organised Tables (IOT) for which the only access is via the index (caveat being if there were secondary indexes on the non-key columns of the IOT we wanted to use such a hint could make sense - but there are not). Specifying an INDEX hint doesn't preclude Oracle from using a non-ideal scan method eg fast full scan, range scan etc) - it just says use this particular index. So these hints provide no utility whatsoever.

2) Where we have USE_NL(t m) LEADING (t m) we are providing a verify specific access pathway that can only be interpreted one way. The CARDINALITY hints are thus redundant (these provide Oracle with expected counts from each table) because they would normally be fed to the optimiser to help it formulate a plan. But since we've stipulated a plan with USE_NL/LEADING, the cardinality is irrelevant.

Reply:
yeah I imagine the `t m` references are probably copied from somewhere, but you're right, they don't make sense here.

Confirmed that:
- we use `ORGANIZATION INDEX` in creating the tables (OracleDdlTable:285)
- on that basis, the `INDEX` hint on the primary key index is redundant
- for this query, there is no join/product to speak of, so `USE_NL` and `LEADING` don't make sense.